src/plugins/saved_objects/public/save_modal/show_saved_object_save_modal.tsx

```diff
@@ -38,6 +38,7 @@ export function showSaveModal(
   const closeModal = () => {
     ReactDOM.unmountComponentAtNode(container);
     document.body.removeChild(container);
+    saveModal.props.onClose?.();
```

Comment:
`showSaveModal` used to just discard the provided modal's `onClose` callback without calling it, but we want to know when the modal's been closed to refocus the menu button.